src/test/blockchain/blockchain_behavior_tests.cpp

```diff
@@ -39,4 +49,92 @@ BOOST_AUTO_TEST_CASE(creation_test) {
   }
 }
 
+BOOST_AUTO_TEST_CASE(GetTransactionWeight_test) {
+  mocks::ArgsManagerMock args{};
+  const auto b = blockchain::Behavior::New(&args);
+
+  {
+    // Check weight of empty transaction
+    CTransaction tx;
+
+    const std::size_t expected = GetTransactionWeight(tx);
+    const std::size_t weight = b->GetTransactionWeight(tx);
+
+    BOOST_CHECK_EQUAL(40, weight);
+    BOOST_CHECK_EQUAL(expected, weight);
+  }
+
+  txtools::TxTool key_tool;
+  {
+    // Check weight of standard transaction
+    const CTransaction tx = key_tool.CreateTransaction();
+
+    const std::size_t expected = GetTransactionWeight(tx);
+    const std::size_t weight = b->GetTransactionWeight(tx);
+
+    BOOST_CHECK_EQUAL(437, weight);
+    BOOST_CHECK_EQUAL(expected, weight);
+  }
+}
+
+BOOST_AUTO_TEST_CASE(GetBlockWeight_test) {
+  mocks::ArgsManagerMock args{};
+  const auto b = blockchain::Behavior::New(&args);
+
+  {
+    // Check weight of empty block
+    CBlock block;
+
+    const std::size_t expected = GetBlockWeight(block);
+    const std::size_t weight = b->GetBlockWeight(block);
+
+    BOOST_CHECK_EQUAL(568, weight);
+    BOOST_CHECK_EQUAL(expected, weight);
+  }
+
+  txtools::TxTool key_tool;
+  {
+    // Check weight of some block
+    CBlock block;
+    const CTransaction tx = key_tool.CreateTransaction();
+    block.vtx.emplace_back(MakeTransactionRef(tx));
+
+    const std::size_t expected = GetBlockWeight(block);
+    const std::size_t weight = b->GetBlockWeight(block);
+
+    BOOST_CHECK_GE(weight, 1004);
```

Comment:
I got 1003 a few times

Reply:
Thank you for exhaustingly checking this. I will adapt.